src/cache.h

```diff
@@ -270,211 +330,541 @@ cache_ctxt_init(cache            *cc,
+ * cache_unlock
+ *
+ * XXX Downgrade a write lock to ... a read-lock with a claim?
+ *
+ *----------------------------------------------------------------------
+ */
 static inline void
 cache_unlock(cache *cc, page_handle *page)
 {
    return cc->ops->page_unlock(cc, page);
 }
 
+/*
+ *----------------------------------------------------------------------
+ * cache_prefetch
+ *
+ * Asynchronously load the extent with given base address. No notification is
+ * provided to the calling thread; it may call cache_get when it's ready
+ * to block on the arrival of the page.
+ *
+ *----------------------------------------------------------------------
+ */
 static inline void
 cache_prefetch(cache *cc, uint64 addr, page_type type)
 {
    return cc->ops->page_prefetch(cc, addr, type);
 }
 
+/*
+ *----------------------------------------------------------------------
+ * cache_mark_dirty
+ *
+ * Marks a page changed, to be written back.
+ * The caller had better have the write lock on the page via cache_lock()
+ * before changing its value.
+ *
+ * XXX Why is this a separate step from cache_lock()? I'd assume that,
+ * by the time a thread upgrades to a write lock, it has already decided
+ * to certainly update the page value. So why add another step to potentially
+ * omit?
+ *
+ *----------------------------------------------------------------------
+ */
 static inline void
 cache_mark_dirty(cache *cc, page_handle *page)
 {
    return cc->ops->page_mark_dirty(cc, page);
 }
 
+/*
+ *----------------------------------------------------------------------
+ * cache_pin
+ *
+ * Pin the page in the cache, disallowing eviction.
+ *
+ * XXX Is this purely a performance optimization: we know this page will
+ * be needed again very soon? Or is this ever used for safety, somehow?
+ * Some weird one-big-lock argument?
+ *
+ * XXX I note that clockcache_evict_all has an !ignore_pinned_pages mode.
+ * Do users of this interface need to know how that interacts?
+ *
+ *----------------------------------------------------------------------
+ */
 static inline void
 cache_pin(cache *cc, page_handle *page)
 {
    return cc->ops->page_pin(cc, page);
 }
 
+/*
+ *----------------------------------------------------------------------
+ * cache_pin
+ *
+ * Release the pin from a cache page, allowing it to be evicted.
+ *
+ * The pin value is reference counted, so multiple threads may pin
+ * and unpin, and the page is only eligible for eviction when no threads
+ * have it pinned.
+ *
+ *----------------------------------------------------------------------
+ */
 static inline void
 cache_unpin(cache *cc, page_handle *page)
 {
    return cc->ops->page_unpin(cc, page);
 }
 
+/*
+ *-----------------------------------------------------------------------------
+ * cache_page_sync
+ *
+ * Asynchronously writes the page back to disk. Currently there is no way to
+ * check when the writeback has completed,
+ * XXX thus this call is only useful to prepare a page for eviction?
```

Comment:
This is used to sync log pages opportunistically. The current API doesn't inform the user when this happens. It's not the ideal API.